compiler/rustc_data_structures/src/sync.rs

```diff
@@ -26,6 +26,36 @@ use std::panic::{catch_unwind, resume_unwind, AssertUnwindSafe};
 pub use std::sync::atomic::Ordering;
 pub use std::sync::atomic::Ordering::SeqCst;
 
+mod mode {
+    use super::Ordering;
+
+    const UNINITIALIZED: usize = 0;
+    const SINGLE_THREAD: usize = 1;
+    const MULTIPLE_THREADS: usize = 2;
+
+    static PARALLEL: std::sync::atomic::AtomicUsize =
+        std::sync::atomic::AtomicUsize::new(UNINITIALIZED);
+
+    #[inline]
+    pub fn is_parallel() -> bool {
+        match PARALLEL.load(Ordering::Relaxed) {
+            SINGLE_THREAD => false,
+            MULTIPLE_THREADS => true,
+            _ => panic!("uninitialized parallel mode!"),
+        }
+    }
+
+    // Only set by the `-Z threads` compile option
+    pub fn set_parallel_mode(threads: usize) {
```

Comment:
This should take a boolean as it doesn't care about the thread count. It would be nice the `-Z threads=-1` option enabled synchronization with a single thread to measure the overhead of synchronization too.

Reply:
Make the parameter to bool. I think we can make a follow-up when enhancing rustc-perf in parallel environment in the future